### ISSUE TITLE
Fix S1481 FP: "var using" declarations should not raise issue

### ISSUE THREAD

**Issue by costin-zaharia-sonarsource** [SonarSource/sonar-dotnet#3137]
### Description

Rule S1481 raises a false positive when the variable is declared inside a using declaration.

### Repro steps

```
void Foo(string path)
{
    using var stream = File.Create(path);
}
```

### Expected behavior

There should be no issue raised.

### Actual behavior

The issue is raised for the variable **stream**

### Known workarounds

```
using(var t = new TestTimer()) // t never used, but compliant.
{
  //...
}
```

but this will raise another issue if the using body is empty.

### Related information

* SonarC# Version 8.3

**Comment by costin-zaharia-sonarsource** [comment]
I'm closing this issue since it would be easier to use `File.Create(path).Dispose();` instead and this will not need a new variable.

**Comment by albankimor** [comment]
Hello,

I encounter the same false positive issue with the following code :

`using var _ = _logger.BeginScope("XXX");`

I think this issue should be reopenned and the rule S1481 should be adapted for the C# 8 syntax.

Thank you for advance

**Comment by pavel-mikula-sonarsource** [comment]
`using var _ = new ...` is not a discard pattern, but `_` variable that could be eventually used. So we should implement this as originally reported by @costin-zaharia-sonarsource

**Comment by GeeWee** [comment]
While `using var _ = new ...` is not an actual discard pattern - I'd like to think the intent is very clear anyways - and that it should not be marked as an error.
